# What is the Natural Language Toolkit (NLTK)?
NLTK is a Python Library for working with written language data. It is free, open source and well documented. The toolkit supports at least 40 different languages and is now used in university courses around the world.

There is an extensive and accessible companion book available free [online](http://www.nltk.org/book/). Many areas covered in this chapter are covered in more detail in the book.

_Note: NLTK provides tools for tasks ranging from very simple (counting words in a text) to very complex (writing and training parsers, etc.). Many advanced tasks are beyond the scope of this chapter_

NLTK is a power tool. Most only use a fraction of its functions for a specific purpose. We will be using NLTK to preprocess and analysis Twitter data. Although we will use methods relevant to Twitter data, some of the methods and techniques used will be applicable to a variety of data. You will learn to analyse this data with some basic textual analytic tools, including counting word frequencies, concordancing and collocational analysis. 

Many other forms of analysis are possible in NLTK, including sentiment analysis. There are beyond the scope of this workshop, but many of the concept learnt here will help you if you wish to extend your knowledge later.



# Install instructions

To follow along with the hands on explanations, please download the following software:

_**Install Python:**_ The first thing you will need for the challenge is to download Python. The easiest way is to download [Anaconda](https://conda.io/miniconda.html). This is not the only way to write Python code, but it is fairly simple and free. Follow the instructions and install the Jupyter Notebook onto your machine. 

_**Install NLTK:**_ Once you've installed the suite of tools that come with Anaconda (which includes the Jupyter Notebook), open the Jupyter Notebook on your machine and install NLTK by typing the following commands into a fresh Jupyter Notebook.


![alt text](/Desktop/nltk_download.png)



You can also follow the instructions on the nltk.org [site](http://www.nltk.org/install.html).



# Textual Analysis with NLTK
Before we dive into processing an analysing your own data, we'll give you can idea of what NLTK can do by working on some pre-processed practice data that comes with NLTK. 

Often it is easier (computationally) to analyse our data, then to upload and pre-process our data. So we'll start back-the-front.

Start by importing NLTK

In [4]:
import nltk

Next, we'll import the example corpora from the nltk book module (the * means 'the lot').

In [5]:
from nltk.book import *

Importing the book has assigned variable names to ten corpora. We can 'call' these names easily:

In [6]:
text2

<Text: Sense and Sensibility by Jane Austen 1811>

Texts ``text1, text2, text3`` etc... are call _**variables**_. In computer programming, variables are data (e.g. “Sense and Sensibility by Jane Austen 1811”) paired with an associated symbolic name or identifier (e.g. 'text2' in the code above). Variables are assigned in python with the ``=`` sign.

Try assigning and calling your own variable. Here's my example below:

In [7]:
my_variable_name = 'the quality of mercy is not strained...'
my_variable_name

'the quality of mercy is not strained...'